### PR TITLE
[device-reachability-status]: support for reachability with multiple connectivity-types

### LINKED ISSUE #140: Extend the possible connectivity values
**Problem description**

Currently we have 3 different values for GET /connectivity 
     - CONNECTED_DATA
     - CONNECTED_SMS
     - NOT_CONNECTED
      
DATA was meant to be returned when data connectivity is given, regardless of the SMS connectivity. But in some cases a finer differentiation is needed.

**Possible evolution**
CONNECTIVITY_DATA should be returned if only data connectivity is given but no SMS.
either a new value CONNECTIVITY_DATA_SMS should be introduced 
or more than 1 value should be returned

The same applies for the connectivity events.

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:
* enhancement/feature

#### What this PR does / why we need it:
Adds a support, when a device is reachable via multiple connectivity-types, e.g. SMS & DATA.

#### Which issue(s) this PR fixes:

Fixes #140

### CODE REVIEW

File: code/Test_definitions/device-reachability-status.feature [hunk 1]
```diff
@@ -13,49 +13,57 @@ Feature: CAMARA Device reachability status API, v0.6.0 - Operations for reachabi
...
     Then the response code is 200
     And the response header "Content-Type" is "application/json"
     And the response header "x-correlator" has same value as the request header "x-correlator"
     And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
-    And the response property "$.status" is 200
-    And the response property "$.code" is "OK"
-    And the response property "$.message" contains a user friendly text
-    And the response property "$.reachabilityStatus" is "CONNECTED_SMS"
+    And the response property "$.reachable" is true
+    And the response property "$.connectivity" is ["SMS"]
```
Comment (eric-murray) [inline code/Test_definitions/device-reachability-status.feature:26]: "a valid testing device which is connected with sms and supported by the service" may also be connected for data. So either we specify that the device should also not be connected for data or, better, just check that "SMS" is in the `connectivity` array.
```suggestion
    And the response property "$.connectivity" includes "SMS"
```

File: code/Test_definitions/device-reachability-status.feature [hunk 2]
```diff
@@ -13,49 +13,57 @@ Feature: CAMARA Device reachability status API, v0.6.0 - Operations for reachabi
...
-    Given a valid devicestatus request body 
+    Given a valid device reachability status request body 
+    And the request body property "$.device" is set to a valid testing device which is connected with data and supported by the service
+    When the request "getReachabilityStatus" is sent
+    Then the response code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
+    And the response property "$.reachable" is true
+    And the response property "$.connectivity" is ["DATA"]
```
Comment (eric-murray) [inline code/Test_definitions/device-reachability-status.feature:38]:
```suggestion
    And the response property "$.connectivity" includes "DATA"
```

File: code/Test_definitions/device-reachability-status.feature [hunk 3]
```diff
@@ -13,49 +13,57 @@ Feature: CAMARA Device reachability status API, v0.6.0 - Operations for reachabi
...
+
+  @device_reachability_status_03_reachableAndConnectedDataAndSms
+  Scenario: Check the reachability status if device is connected with DATA
```
Comment (eric-murray) [inline code/Test_definitions/device-reachability-status.feature:41]:
```suggestion
  Scenario: Check the reachability status if device is connected with DATA and SMS
```

File: code/Test_definitions/device-reachability-status.feature [hunk 4]
```diff
@@ -13,49 +13,57 @@ Feature: CAMARA Device reachability status API, v0.6.0 - Operations for reachabi
...
     When the request "getReachabilityStatus" is sent
     Then the response code is 200
     And the response header "Content-Type" is "application/json"
     And the response header "x-correlator" has same value as the request header "x-correlator"
     And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
-    And the response property "$.status" is 200
-    And the response property "$.code" is "OK"
-    And the response property "$.reachabilityStatus" is  "CONNECTED_DATA"
+    And the response property "$.reachable" is true
+    And the response property "$.connectivity" is ["DATA", "SMS"]
```
Comment (eric-murray) [inline code/Test_definitions/device-reachability-status.feature:50]: We should not exclude ["SMS", "DATA"] as a valid response!
```suggestion
    And the response property "$.connectivity" includes both "DATA" and "SMS"
```

**Review by eric-murray — DISMISSED** [review]

**Review by bigludo7 — COMMENTED** [review]
Line 78 should be `  - url: "{apiRoot}/device-reachability-status/vwip"` no?

**Comment by akoshunyadi** [comment]
Please don't merge until the patch release is out! Thanks!

**Review by eric-murray — APPROVED** [review]

**Review by sachinvodafone — APPROVED** [review]

**Comment by maxl2287** [comment]
@akoshunyadi & @bigludo7 I guess we can merge it now, agree?

**Comment by akoshunyadi** [comment]
yes, now the main branch is free for development

**Pull request merged**